### PR TITLE
Get children files and folders independently

### PR DESCRIPTION
**Pull request by sergiohs84**
A new `MegaChildrenLists` wrapper has been created in order to avoid two passes through the children vector, which would be required in case of two different APIs for files and folders.

### CODE REVIEW

File: bindings/wp8/MChildrenLists.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+MChildrenLists^ MChildrenLists::copy()
+{
+    return childrenLists ? ref new MChildrenLists(childrenLists->copy(), true) : nullptr;
+}
+
+MNodeList^ MChildrenLists::getFolderList()
+{
+    return childrenLists ? ref new MNodeList(childrenLists->getFolderList(), true) : nullptr;
```
Comment (javiserrano) [inline bindings/wp8/MChildrenLists.cpp]: `getFolderList()` doesn't give the ownership of the returned list, so the second parameter should be `false` (and the app should take care to not use the returned `MNodeList` after deleting the `MChildrenList`) or a  copy of the list should be created to pass it to the constructor. The second is more inefficient but also more secure.

File: bindings/wp8/MChildrenLists.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+MNodeList^ MChildrenLists::getFileList()
+{
+    return childrenLists ? ref new MNodeList(childrenLists->getFileList(), true) : nullptr;
```
Comment (javiserrano) [inline bindings/wp8/MChildrenLists.cpp]: `getFileList()` doesn't give the ownership of the returned list, so the second parameter should be `false` (and the app should take care to not use the returned `MNodeList` after deleting the `MChildrenList`) or a  copy of the list should be created to pass it to the constructor. The second is more inefficient but also more secure.

File: bindings/wp8/MChildrenLists.cpp (hunk 1 above)
Comment (javiergm1983) [inline bindings/wp8/MChildrenLists.cpp]: Requested changes done.

File: bindings/wp8/MChildrenLists.cpp (hunk 2 above)
Comment (javiergm1983) [inline bindings/wp8/MChildrenLists.cpp]: Requested changes done.

**Pull request merged**